cupy/lib/polynomial.pyx

```diff
@@ -0,0 +1,275 @@
+import re
+import warnings
+
+import numpy
+
+import cupy
+
+_poly_mat = re.compile(r'[*][*]([0-9]*)')
+
+
+def _raise_power(astr, wrap=70):
+    n = 0
+    line1 = ''
+    line2 = ''
+    output = ' '
+    while True:
+        mat = _poly_mat.search(astr, n)
+        if mat is None:
+            break
+        span = mat.span()
+        power = mat.groups()[0]
+        partstr = astr[n:span[0]]
+        n = span[1]
+        toadd2 = partstr + ' ' * (len(power) - 1)
+        toadd1 = ' ' * (len(partstr) - 1) + power
+        if ((len(line2) + len(toadd2) > wrap) or
+                (len(line1) + len(toadd1) > wrap)):
+            output += line1 + '\n' + line2 + '\n '
+            line1 = toadd1
+            line2 = toadd2
+        else:
+            line2 += partstr + ' ' * (len(power) - 1)
+            line1 += ' ' * (len(partstr) - 1) + power
+    output += line1 + '\n' + line2
+    return output + astr[n:]
+
+
+class poly1d(object):
+    """A one-dimensional polynomial class.
+
+    Args:
+        c_or_r (cupy.ndarray or cupy.poly1d): The polynomial's
```

Comment:
There is no interface for conversion from `numpy.poly1d` to `cupy.poly1d` currently. How about supporting it here?

Reply:
I think it's OK to allow all `array_like` objects like `cupy.array` in this case.